**Manual test on MacOS**

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ uname
Darwin
```

Ensure no native binaries are built beforehand.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ ll dist/
total 0
drwxr-xr-x   2 felixsatyaputra  staff    68B 23 Mar 10:56 .
drwxr-xr-x  36 felixsatyaputra  staff   1.2K 23 Mar 10:56 ..
```

Run the script, giving `1.1.0` as the input to match the current CLI version specified in `conductr_cli/__init__.py`.

The script will build the native executables, validate its version, validates if the commands are working properly by running:

* `sandbox run 2.0.0 -f visualization`
* `conduct info`
* `shazar -h`

The Visualizer being started will be validated using `curl http://192.168.10.1:9008/services/visualizer`.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ cat conductr_cli/__init__.py
__version__ = '1.1.0'

Felixs-MBP-2:conductr-cli felixsatyaputra$ sh package-native-zip.sh 1.1.0

Building conductr-cli-1.1.0-Mac_OS_X-x86_64.zip

------------------------------------------------
Creating single executable for 'conduct' command
------------------------------------------------

81 INFO: PyInstaller: 3.2.1
81 INFO: Python: 3.5.2
88 INFO: Platform: Darwin-15.3.0-x86_64-i386-64bit
88 INFO: wrote /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conduct.spec
90 INFO: UPX is not available.
91 INFO: Extending PYTHONPATH with paths
['/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli',
 '/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli']
91 INFO: checking Analysis
114 INFO: Building because /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/constants.py changed
114 INFO: Initializing module dependency graph...
116 INFO: Initializing module graph hooks...
117 INFO: Analyzing base_library.zip ...
2562 INFO: running Analysis out00-Analysis.toc
2569 INFO: Caching module hooks...
2574 INFO: Analyzing /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct.py
4208 INFO: Processing pre-safe import module hook   six.moves
4521 INFO: Processing pre-safe import module hook   win32com
Traceback (most recent call last):
  File "<string>", line 2, in <module>
ImportError: No module named 'win32com'
4552 INFO: Processing pre-safe import module hook   win32com
Traceback (most recent call last):
  File "<string>", line 2, in <module>
ImportError: No module named 'win32com'
4610 INFO: Processing pre-find module path hook   distutils
6099 INFO: Loading module hooks...
6099 INFO: Loading module hook "hook-requests.py"...
6102 INFO: Loading module hook "hook-pkg_resources.py"...
6103 INFO: Loading module hook "hook-encodings.py"...
6163 INFO: Loading module hook "hook-sysconfig.py"...
6167 INFO: Loading module hook "hook-lib2to3.py"...
6172 INFO: Loading module hook "hook-xml.py"...
6232 INFO: Loading module hook "hook-jsonschema.py"...
6233 INFO: Loading module hook "hook-distutils.py"...
6235 INFO: Loading module hook "hook-pydoc.py"...
6259 INFO: Looking for ctypes DLLs
6274 WARNING: library user32 required via ctypes not found
6274 INFO: Analyzing run-time hooks ...
6280 INFO: Including run-time hook 'pyi_rth_pkgres.py'
6288 INFO: Looking for dynamic libraries
6329 ERROR: Can not find path /usr/local/opt/readline/lib/libreadline.6.dylib (needed by /usr/local/Cellar/python3/3.5.2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/lib-dynload/readline.cpython-35m-darwin.so)
6373 INFO: Looking for eggs
6373 INFO: Using Python library /usr/local/Cellar/python3/3.5.2/Frameworks/Python.framework/Versions/3.5/Python
6380 INFO: Warnings written to /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/build/conduct/warnconduct.txt
6399 INFO: checking PYZ
6404 INFO: Building because toc changed
6404 INFO: Building PYZ (ZlibArchive) /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/build/conduct/out00-PYZ.pyz
7008 INFO: Building PYZ (ZlibArchive) /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/build/conduct/out00-PYZ.pyz completed successfully.
7021 INFO: checking PKG
7023 INFO: Building because toc changed
7023 INFO: Building PKG (CArchive) out00-PKG.pkg
8754 INFO: Building PKG (CArchive) out00-PKG.pkg completed successfully.
8757 INFO: Bootloader /usr/local/lib/python3.5/site-packages/PyInstaller/bootloader/Darwin-64bit/run
8757 INFO: checking EXE
8758 INFO: Rebuilding out00-EXE.toc because conduct missing
8758 INFO: Building EXE from out00-EXE.toc
8758 INFO: Appending archive to EXE /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/dist/conduct
8763 INFO: Fixing EXE for code signing /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/dist/conduct
8768 INFO: Building EXE from out00-EXE.toc completed successfully.
------------------------------------------------
Creating single executable for 'shazar' command
------------------------------------------------

36 INFO: PyInstaller: 3.2.1
37 INFO: Python: 3.5.2
40 INFO: Platform: Darwin-15.3.0-x86_64-i386-64bit
41 INFO: wrote /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/shazar.spec
43 INFO: UPX is not available.
44 INFO: Extending PYTHONPATH with paths
['/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli',
 '/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli']
44 INFO: checking Analysis
48 INFO: Building because /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/constants.py changed
48 INFO: Initializing module dependency graph...
49 INFO: Initializing module graph hooks...
50 INFO: Analyzing base_library.zip ...
2287 INFO: running Analysis out00-Analysis.toc
2292 INFO: Caching module hooks...
2294 INFO: Analyzing /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/shazar.py
2479 INFO: Loading module hooks...
2479 INFO: Loading module hook "hook-pydoc.py"...
2480 INFO: Loading module hook "hook-xml.py"...
2667 INFO: Loading module hook "hook-encodings.py"...
2744 INFO: Looking for ctypes DLLs
2751 INFO: Analyzing run-time hooks ...
2757 INFO: Looking for dynamic libraries
2768 ERROR: Can not find path /usr/local/opt/readline/lib/libreadline.6.dylib (needed by /usr/local/Cellar/python3/3.5.2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/lib-dynload/readline.cpython-35m-darwin.so)
2802 INFO: Looking for eggs
2802 INFO: Using Python library /usr/local/Cellar/python3/3.5.2/Frameworks/Python.framework/Versions/3.5/Python
2805 INFO: Warnings written to /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/build/shazar/warnshazar.txt
2813 INFO: checking PYZ
2815 INFO: Building because toc changed
2815 INFO: Building PYZ (ZlibArchive) /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/build/shazar/out00-PYZ.pyz
3125 INFO: Building PYZ (ZlibArchive) /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/build/shazar/out00-PYZ.pyz completed successfully.
3131 INFO: checking PKG
3132 INFO: Building because toc changed
3132 INFO: Building PKG (CArchive) out00-PKG.pkg
4666 INFO: Building PKG (CArchive) out00-PKG.pkg completed successfully.
4668 INFO: Bootloader /usr/local/lib/python3.5/site-packages/PyInstaller/bootloader/Darwin-64bit/run
4668 INFO: checking EXE
4671 INFO: Rebuilding out00-EXE.toc because shazar missing
4671 INFO: Building EXE from out00-EXE.toc
4671 INFO: Appending archive to EXE /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/dist/shazar
4675 INFO: Fixing EXE for code signing /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/dist/shazar
4677 INFO: Building EXE from out00-EXE.toc completed successfully.
------------------------------------------------
Creating single executable for 'sandbox' command
------------------------------------------------

37 INFO: PyInstaller: 3.2.1
37 INFO: Python: 3.5.2
41 INFO: Platform: Darwin-15.3.0-x86_64-i386-64bit
41 INFO: wrote /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/sandbox.spec
43 INFO: UPX is not available.
44 INFO: Extending PYTHONPATH with paths
['/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli',
 '/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli']
44 INFO: checking Analysis
53 INFO: Building because /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/sandbox_run_jvm.py changed
53 INFO: Initializing module dependency graph...
54 INFO: Initializing module graph hooks...
55 INFO: Analyzing base_library.zip ...
2225 INFO: Analyzing hidden import 'psutil'
2448 INFO: running Analysis out00-Analysis.toc
2453 INFO: Caching module hooks...
2455 INFO: Analyzing /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/sandbox.py
3971 INFO: Processing pre-safe import module hook   six.moves
4644 INFO: Processing pre-safe import module hook   win32com
Traceback (most recent call last):
  File "<string>", line 2, in <module>
ImportError: No module named 'win32com'
4679 INFO: Processing pre-safe import module hook   win32com
Traceback (most recent call last):
  File "<string>", line 2, in <module>
ImportError: No module named 'win32com'
4737 INFO: Processing pre-find module path hook   distutils
5789 INFO: Loading module hooks...
5789 INFO: Loading module hook "hook-sysconfig.py"...
5790 INFO: Loading module hook "hook-xml.py"...
5848 INFO: Loading module hook "hook-distutils.py"...
5848 INFO: Loading module hook "hook-encodings.py"...
5913 INFO: Loading module hook "hook-pydoc.py"...
5914 INFO: Loading module hook "hook-jsonschema.py"...
5915 INFO: Loading module hook "hook-pkg_resources.py"...
5915 INFO: Loading module hook "hook-xml.etree.cElementTree.py"...
5918 INFO: Loading module hook "hook-requests.py"...
5920 INFO: Loading module hook "hook-lib2to3.py"...
5946 INFO: Looking for ctypes DLLs
5958 WARNING: library user32 required via ctypes not found
5960 INFO: Analyzing run-time hooks ...
5966 INFO: Including run-time hook 'pyi_rth_pkgres.py'
5976 INFO: Looking for dynamic libraries
5998 ERROR: Can not find path /usr/local/opt/readline/lib/libreadline.6.dylib (needed by /usr/local/Cellar/python3/3.5.2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/lib-dynload/readline.cpython-35m-darwin.so)
6041 INFO: Looking for eggs
6041 INFO: Using Python library /usr/local/Cellar/python3/3.5.2/Frameworks/Python.framework/Versions/3.5/Python
6046 INFO: Warnings written to /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/build/sandbox/warnsandbox.txt
6062 INFO: checking PYZ
6068 INFO: Building because toc changed
6068 INFO: Building PYZ (ZlibArchive) /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/build/sandbox/out00-PYZ.pyz
6708 INFO: Building PYZ (ZlibArchive) /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/build/sandbox/out00-PYZ.pyz completed successfully.
6723 INFO: checking PKG
6724 INFO: Building because toc changed
6724 INFO: Building PKG (CArchive) out00-PKG.pkg
8412 INFO: Building PKG (CArchive) out00-PKG.pkg completed successfully.
8415 INFO: Bootloader /usr/local/lib/python3.5/site-packages/PyInstaller/bootloader/Darwin-64bit/run
8415 INFO: checking EXE
8417 INFO: Rebuilding out00-EXE.toc because sandbox missing
8417 INFO: Building EXE from out00-EXE.toc
8417 INFO: Appending archive to EXE /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/dist/sandbox
8422 INFO: Fixing EXE for code signing /Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/dist/sandbox
8424 INFO: Building EXE from out00-EXE.toc completed successfully.
------------------------------------------------
Validating version for 'conduct' command
------------------------------------------------

------------------------------------------------
Validating version for 'sandbox' command
------------------------------------------------

------------------------------------------------
Checking 'sandbox' and 'conduct' command is working as expected
------------------------------------------------

|------------------------------------------------|
| Starting ConductR                              |
|------------------------------------------------|
Extracting ConductR core to /Users/felixsatyaputra/.conductr/images/core
Extracting ConductR agent to /Users/felixsatyaputra/.conductr/images/agent
Starting ConductR core instance on 192.168.10.1..
Waiting for ConductR to start..
Starting ConductR agent instance on 192.168.10.1..
|------------------------------------------------|
| Starting proxying feature                      |
|------------------------------------------------|
HAProxy has not been started
To enable proxying ensure Docker is running and restart the sandbox
|------------------------------------------------|
| Starting logging feature based on eslite       |
|------------------------------------------------|
Deploying bundle eslite..
Retrieving bundle..
Loading bundle from cache typesafe/bundle/eslite
Bintray credentials loaded from /Users/felixsatyaputra/.lightbend/commercial.credentials
Retrieving from cache /Users/felixsatyaputra/.conductr/cache/bundle/eslite-v1-3349b6bfbc09e8d88123fee5b33b2717e07fb2a91de719c99e0403c9e4b28ea5.zip
Loading bundle to ConductR..
[#################################################] 100%
Bundle 3349b6bfbc09e8d88123fee5b33b2717 is installed
Bundle loaded.
Bundle run request sent.
Bundle 3349b6bfbc09e8d88123fee5b33b2717 waiting to reach expected scale 1
Bundle 3349b6bfbc09e8d88123fee5b33b2717 has scale 0, expected 1...
Bundle 3349b6bfbc09e8d88123fee5b33b2717 expected scale 1 is met
|------------------------------------------------|
| Starting visualization feature                 |
|------------------------------------------------|
Deploying bundle visualizer..
Retrieving bundle..
Loading bundle from cache typesafe/bundle/visualizer
Bintray credentials loaded from /Users/felixsatyaputra/.lightbend/commercial.credentials
Retrieving from cache /Users/felixsatyaputra/.conductr/cache/bundle/visualizer-v2-73595ecbdad1f01a05db5304046b4ad5c0a98ef83e201e507a153cea21087078.zip
Loading bundle to ConductR..
[#################################################] 100%
Bundle 73595ecbdad1f01a05db5304046b4ad5 is installed
Bundle loaded.
Bundle run request sent.
Bundle 73595ecbdad1f01a05db5304046b4ad5 waiting to reach expected scale 1
Bundle 73595ecbdad1f01a05db5304046b4ad5 has scale 0, expected 1.
Bundle 73595ecbdad1f01a05db5304046b4ad5 expected scale 1 is met
|------------------------------------------------|
| Summary                                        |
|------------------------------------------------|
|- - - - - - - - - - - - - - - - - - - - - - - - |
| ConductR                                       |
|- - - - - - - - - - - - - - - - - - - - - - - - |
ConductR has been started:
  core instance on 192.168.10.1
  agent instance on 192.168.10.1
ConductR service locator has been started on:
  192.168.10.1:9008
|- - - - - - - - - - - - - - - - - - - - - - - - |
| Features                                       |
|- - - - - - - - - - - - - - - - - - - - - - - - |
The following feature related bundles have been started:
  visualizer on 192.168.10.1:9008/services/visualizer
|- - - - - - - - - - - - - - - - - - - - - - - - |
| Bundles                                        |
|- - - - - - - - - - - - - - - - - - - - - - - - |
Check latest bundle status with:
  conduct info
Current bundle status:
ID       NAME        VER  #REP  #STR  #RUN  ROLES
3349b6b  eslite       v1     1     0     1  elasticsearch
73595ec  visualizer   v2     1     0     1  web
Checking Visualizer
Located at http://192.168.10.1:10917
ID       NAME        VER  #REP  #STR  #RUN  ROLES
3349b6b  eslite       v1     1     0     1  elasticsearch
73595ec  visualizer   v2     1     0     1  web
|------------------------------------------------|
| Stopping ConductR                              |
|------------------------------------------------|
ConductR core pid 92282 stopped
ConductR agent pid 92383 stopped
ConductR has been successfully stopped
------------------------------------------------
Checking 'shazar' command is working as expected
------------------------------------------------

usage: shazar [-h] [-o [OUTPUT]] [--output-dir OUTPUT_DIR] [--tar] [source]

Package a bundle directory or bundle configuration file

positional arguments:
  source                Optional path to a bundle directory or bundle
                        configuration file or tar file.

optional arguments:
  -h, --help            show this help message and exit
  -o [OUTPUT], --output [OUTPUT]
                        The target output file
  --output-dir OUTPUT_DIR
                        When provided with a directory name to package, the
                        directory to write the bundle to, defaults to '.'
  --tar                 If provided, source is decoded as a tar file
------------------------------------------------
Building zip archive for conductr-cli-1.1.0-Mac_OS_X-x86_64.zip
------------------------------------------------

  adding: conduct (deflated 2%)
  adding: sandbox (deflated 2%)
  adding: shazar (deflated 1%)
------------------------------------------------
Success
------------------------------------------------
Created archive in dist/conductr-cli-1.1.0-Mac_OS_X-x86_64.zip
```

The archive and binary is created successfully.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ ll dist/
total 66624
drwxr-xr-x   6 felixsatyaputra  staff   204B 23 Mar 10:57 .
drwxr-xr-x  36 felixsatyaputra  staff   1.2K 23 Mar 10:56 ..
-rwxr-xr-x   1 felixsatyaputra  staff   5.9M 23 Mar 10:56 conduct
-rw-r--r--   1 felixsatyaputra  staff    16M 23 Mar 10:57 conductr-cli-1.1.0-Mac_OS_X-x86_64.zip
-rwxr-xr-x   1 felixsatyaputra  staff   6.1M 23 Mar 10:57 sandbox
-rwxr-xr-x   1 felixsatyaputra  staff   4.4M 23 Mar 10:56 shazar
```

If no input is given, the script stops and failed with an error.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ sh package-native-zip.sh
ERROR: Package version number must be specified

Felixs-MBP-2:conductr-cli felixsatyaputra$ echo $?
1
```

--- a/README.rst
+++ b/README.rst
@@ -289,12 +289,41 @@ For Linux:
 For Windows:
 - Windows 7
 - Python version ``3.5``.
+- 7Zip executable from http://www.7-zip.org/download.html required to build zip file on the command line. When installing 7Zip, ensure the 7z.exe is available on the Windows ``%PATH%``.
 
 Pyinstaller version ``3.2.1`` or above is required. Please visit http://www.pyinstaller.org/ to obtain instructions on how to install it.
 
 Ensure the native executables are built from tagged release commit.
 
-To build native packages follow the form:
+Continue to build native packages.
+
+For MacOS and Linux:
+
+.. code:: bash
+
+    sh package-native-zip.sh [release-version-number]
+
+
+For Windows perform the following steps.
+
+Open a DOS prompt, and then execute the following command.
+
+.. code:: bash
+
+    set CONDUCTR_HOST=192.168.10.1
+
+For those using Windows VM, the local sandbox address ``192.168.10.1`` can be used - ensure the sandbox on the host machine has been started before proceeding further. This will allow the CLI on the Windows VM to connect to the ConductR running on the host machine.
+
+If you wish to use ConductR running from a different host, replace ``192.168.10.1`` accordingly.
+
+.. code:: bash
+
+    package-native-zip.bat [release-version-number]
+
+
+The ``package-native-zip.sh`` and ``package-native-zip.bat`` follow performs the following steps.
+
+First it builds the native executables.
 
 .. code:: bash
 
@@ -304,7 +333,7 @@ To build native packages follow the form:
     
 This will result in standalone images for your current environment being created in a ``dist`` folder.
 
-Ensure correct versions are built.
+It will ensure correct versions are built. This is done by comparing the version number from the output of the commands below with the input to the script. If there's a mismatch, the script will exit with failure.
 
 .. code:: bash
 
@@ -323,7 +352,7 @@ For Windows, perform the following since the ``sandbox`` command is not supporte
 
 .. code:: bash
 
-    ./dist/conduct info --host [host of some running ConductR]
+    ./dist/conduct info
     ./dist/shazar -h
 
 

--- a/package-native-zip.bat
+++ b/package-native-zip.bat
@@ -1,0 +1,82 @@
+echo off
+SETLOCAL
+
+set PACKAGE_VERSION_NUMBER=%1
+
+if [%PACKAGE_VERSION_NUMBER%] == [] (
+    echo "ERROR: Package version number must be specified"
+    goto EndWithError
+)
+
+del /q dist\*
+
+set PACKAGE_OS_NAME=Win32
+set PACKAGE_NAME=conductr-cli-%PACKAGE_VERSION_NUMBER%-%PACKAGE_OS_NAME%.zip
+
+echo.
+echo Building %PACKAGE_NAME%
+echo.
+
+echo ------------------------------------------------
+echo Creating single executable for 'conduct' command
+echo ------------------------------------------------
+echo.
+pyinstaller --onefile conductr_cli/conduct.py
+
+echo ------------------------------------------------
+echo Creating single executable for 'shazar' command
+echo ------------------------------------------------
+echo.
+pyinstaller --onefile conductr_cli/shazar.py
+
+echo ------------------------------------------------
+echo Creating single executable for 'sandbox' command
+echo ------------------------------------------------
+echo.
+pyinstaller --hidden-import psutil --onefile conductr_cli/sandbox.py
+
+echo ------------------------------------------------
+echo Validating version for 'conduct' command
+echo ------------------------------------------------
+echo.
+del c:\temp\conduct-version.txt
+conduct version > c:\temp\conduct-version.txt
+set /p CONDUCT_VERSION=< c:\temp\conduct-version.txt
+
+if %CONDUCT_VERSION% NEQ %PACKAGE_VERSION_NUMBER% (
+    echo ERROR: Mismatched version number for 'conduct' command
+    echo ERROR: 'conduct' command version: %CONDUCT_VERSION%
+    echo ERROR: Package version: %PACKAGE_VERSION_NUMBER%
+    goto EndWithError
+)
+
+echo ------------------------------------------------
+echo Checking 'conduct' command is working as expected
+echo ------------------------------------------------
+echo.
+conduct info
+
+echo ------------------------------------------------
+echo Checking 'shazar' command is working as expected
+echo ------------------------------------------------
+echo.
+shazar -h
+
+
+echo ------------------------------------------------
+echo Building zip archive for %PACKAGE_NAME%
+echo ------------------------------------------------
+echo.
+cd dist
+7z a %PACKAGE_NAME% conduct.exe shazar.exe sandbox.exe
+cd ..
+
+echo ------------------------------------------------
+echo Success
+echo ------------------------------------------------
+echo Created archive in dist/%PACKAGE_NAME%
+echo.
+echo.
+
+:EndWithError
+echo.

--- a/package-native-zip.sh
+++ b/package-native-zip.sh
@@ -1,0 +1,112 @@
+#!/bin/bash
+
+set -e
+
+CURRENT_DIR=`pwd`
+PACKAGE_VERSION_NUMBER="$1"
+
+if [ -z "$PACKAGE_VERSION_NUMBER" ]; then
+    echo "ERROR: Package version number must be specified"
+    exit 1
+fi
+
+UNAME_VALUE=`uname`
+
+if [ "$UNAME_VALUE" = "Darwin" ]; then
+    PACKAGE_OS_NAME="Mac_OS_X-x86_64"
+elif [ "$UNAME_VALUE" = "Linux" ]; then
+    PACKAGE_OS_NAME="Linux-amd64"
+else
+    echo "ERROR: Unable to build package for $UNAME_VALUE"
+    exit 1
+fi
+
+PACKAGE_NAME="conductr-cli-$PACKAGE_VERSION_NUMBER-$PACKAGE_OS_NAME.zip"
+
+echo ""
+echo "Building $PACKAGE_NAME"
+echo ""
+
+rm -f dist/*
+
+echo "------------------------------------------------"
+echo "Creating single executable for 'conduct' command"
+echo "------------------------------------------------"
+echo ""
+pyinstaller --onefile conductr_cli/conduct.py
+
+echo "------------------------------------------------"
+echo "Creating single executable for 'shazar' command"
+echo "------------------------------------------------"
+echo ""
+pyinstaller --onefile conductr_cli/shazar.py
+
+echo "------------------------------------------------"
+echo "Creating single executable for 'sandbox' command"
+echo "------------------------------------------------"
+echo ""
+pyinstaller --hidden-import psutil --onefile conductr_cli/sandbox.py
+
+
+echo "------------------------------------------------"
+echo "Validating version for 'conduct' command"
+echo "------------------------------------------------"
+echo ""
+# Do not use `head -n 1` to trim the first line of the output as this will result in broken pipe error in Linux.
+CONDUCT_VERSION=$(dist/conduct version)
+CONDUCT_VERSION=$(echo $CONDUCT_VERSION | awk '{print $1}')
+
+if [ "$CONDUCT_VERSION" != "$PACKAGE_VERSION_NUMBER" ]; then
+    echo "ERROR: Mismatched version number for 'conduct' command"
+    echo "ERROR: 'conduct' command version: $CONDUCT_VERSION"
+    echo "ERROR: Package version: $PACKAGE_VERSION_NUMBER"
+    exit 1
+fi
+
+echo "------------------------------------------------"
+echo "Validating version for 'sandbox' command"
+echo "------------------------------------------------"
+echo ""
+# Do not use `head -n 1` to trim the first line of the output as this will result in broken pipe error in Linux.
+SANDBOX_VERSION=$(dist/sandbox version)
+SANDBOX_VERSION=$(echo $SANDBOX_VERSION | awk '{print $1}')
+
+if [ "$SANDBOX_VERSION" != "$PACKAGE_VERSION_NUMBER" ]; then
+    echo "ERROR: Mismatched version number for 'sandbox' command"
+    echo "ERROR: 'sandbox' command version: $SANDBOX_VERSION"
+    echo "ERROR: Package version: $PACKAGE_VERSION_NUMBER"
+    exit 1
+fi
+
+echo "------------------------------------------------"
+echo "Checking 'sandbox' and 'conduct' command is working as expected"
+echo "------------------------------------------------"
+echo ""
+
+dist/sandbox run 2.0.0 -f visualization
+echo "Checking Visualizer"
+curl http://192.168.10.1:9008/services/visualizer
+echo ""
+conduct info
+dist/sandbox stop
+
+echo "------------------------------------------------"
+echo "Checking 'shazar' command is working as expected"
+echo "------------------------------------------------"
+echo ""
+
+dist/shazar -h
+
+echo "------------------------------------------------"
+echo "Building zip archive for $PACKAGE_NAME"
+echo "------------------------------------------------"
+echo ""
+zip -j dist/$PACKAGE_NAME dist/conduct dist/sandbox dist/shazar
+
+echo "------------------------------------------------"
+echo "Success"
+echo "------------------------------------------------"
+echo "Created archive in dist/$PACKAGE_NAME"
+echo ""
+echo ""
+

